lib/ts/recipe/emailverification/index.ts

```diff
@@ -37,26 +65,68 @@ export default class Wrapper {
         });
     }
 
-    static async isEmailVerified(userId: string, email: string, userContext?: any) {
-        return await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.isEmailVerified({
+    static async isEmailVerified(userId: string, email?: string, userContext?: any) {
+        const recipeInstance = Recipe.getInstanceOrThrowError();
+        if (email === undefined) {
+            const emailInfo = await recipeInstance.getEmailForUserId(userId, userContext);
+
+            if (emailInfo.status === "OK") {
+                email = emailInfo.email;
+            } else if (emailInfo.status === "EMAIL_DOES_NOT_EXIST_ERROR") {
+                return true;
+            } else {
+                throw new global.Error("Unknown User ID provided without email");
+            }
+        }
+
+        return await recipeInstance.recipeInterfaceImpl.isEmailVerified({
             userId,
-            email,
+            email: email!,
             userContext: userContext === undefined ? {} : userContext,
         });
     }
 
-    static async revokeEmailVerificationTokens(userId: string, email: string, userContext?: any) {
-        return await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.revokeEmailVerificationTokens({
+    static async revokeEmailVerificationTokens(userId: string, email?: string, userContext?: any) {
+        const recipeInstance = Recipe.getInstanceOrThrowError();
+
+        if (email === undefined) {
+            const emailInfo = await recipeInstance.getEmailForUserId(userId, userContext);
+            if (emailInfo.status === "OK") {
+                email = emailInfo.email;
+            } else if (emailInfo.status === "EMAIL_DOES_NOT_EXIST_ERROR") {
+                return {
+                    status: "EMAIL_ALREADY_VERIFIED",
+                };
```

Comment:
Shouldn't this be returning status: "OK"? Cause the recipe implementation of this does not return a type of `EMAIL_ALREADY_VERIFIED`

Reply:
Hmm, I think throwing would match the old behavior better.